src/RequestInterface.ts

```diff
@@ -34,7 +34,14 @@ export interface RequestInterface<D extends object = object> {
    * @param {string} route Request method + URL. Example: `'GET /orgs/{org}'`
    * @param {object} [parameters] URL, query or body parameters, as well as `headers`, `mediaType.{format|previews}`, `request`, or `baseUrl`.
    */
-  (route: Route, options?: RequestParameters): Promise<OctokitResponse<any>>;
+  <R extends Route>(
+    route: keyof Endpoints | R,
+    options?: R extends keyof Endpoints
+      ? Endpoints[R]["parameters"] & RequestParameters
+      : RequestParameters
+  ): R extends keyof Endpoints
+    ? Promise<Endpoints[R]["response"]>
+    : Promise<OctokitResponse<any>>;
```

Comment:
This is not a full revert of the changes at https://github.com/octokit/types.ts/pull/399/files, is that on purpose?

Reply:
Yes that's on purpose. The first overload fixes routes autocompletion in TS 4.7, but we still need the old one for options autocomplete to work properly.

Reply:
We reverted #399 via #405 because it had some more unforeseen consequences. I'd prefer we would figure out what broke with TS 4.7 and help the team to fix the problem on their side